plugins/out_stackdriver/stackdriver.c

```diff
@@ -2249,6 +2263,15 @@ static flb_sds_t stackdriver_format(struct flb_stackdriver *ctx,
             flb_sds_destroy(trace);
         }
 
+        /* Add spanId field into the log entry */
+        if (span_id_extracted == FLB_TRUE) {
+            msgpack_pack_str(&mp_pck, 6);
+            msgpack_pack_str_body(&mp_pck, "spanId", 6);
+            len = flb_sds_len(span_id);
+            msgpack_pack_str(&mp_pck, len);
+            msgpack_pack_str_body(&mp_pck, span_id, len);
```

Comment:
I think you'll need a `flb_sds_destroy` for the `span_id` string here as well. See the above code where `trace` is added into the log entry; you'll just need to do the destroy at the end of this `if` block.

After this change, if you could do the same valgrind as you did earlier that would be great, just to verify this is resolved.

Reply:
Just as a tip, when you are packing an entire string in `msgpack-c` you can use `msgpack_pack_str_with_body` which packs the string header and body in a single operation.

Reply:
That's great to know, we probably should be doing this all over this plugin, since we currently do it in two steps every time right now. I'll try to do that as another PR some time soon.

@andreswebs for this PR, could you just make this function use that function? You'll be changing from this:

```
msgpack_pack_str(&mp_pck, len);
msgpack_pack_str_body(&mp_pck, span_id, len);
```
To
```
msgpack_pack_str_with_body(&mp_pck, span_id, len);
```
For both packing the `spanId` key and the `span_id` value.